opm/parser/eclipse/EclipseState/SimulationConfig/HysteresisOptions.cpp

```diff
@@ -0,0 +1,230 @@
+#include <opm/parser/eclipse/EclipseState/SimulationConfig/HysteresisOptions.hpp>
+#include <opm/parser/eclipse/Utility/String.hpp>
+
+namespace Opm {
+
+namespace {
+
+HysteresisOptions::Model readHysteresisModel( int x ) {
+    using Model = HysteresisOptions::Model;
+    switch( x ) {
+        case -1: return Model::Equilibration;
+        case  0: return Model::CarlsonHysteresisNonWettingDrainageWetting;
+        case  1: return Model::CarlsonHysteresisNonWettingImbibitionNonWetting;
+        case  2: return Model::KilloughHysteresisNonWettingDrainageWetting;
+        case  3: return Model::KilloughHysteresisNonWettingImbibitionWetting;
+        case  4: return Model::KilloughHysteresisNonWettingKilloughHysteresisWetting;
+        case  5: return Model::CarlsonNonWettingGasWaterDrainageOil;
+        case  6: return Model::KilloughNonWettingGasWaterDrainageOil;
+        case  7: return Model::KilloughHysteresisNonWettingGasWaterWettingOil;
+        case  8: return Model::JargonHysteresisNonWettingDrainageWetting;
+        case  9: return Model::JargonHysteresisNonWettingImbibitionWetting;
```

Comment:
I guess there's quite some work left for opm-material to implement all of these ;)